include/deal.II/fe/fe_interface_values.h

```diff
@@ -0,0 +1,686 @@
+                    const UpdateFlags                   update_flags);
+
+  /**
+   * Re-initialize this object to be used on a new interface given by two faces
+   * of two neighboring cells.
+   *
+   * The arguments (including their order) is identical to the @p face_worker arguments
+   * in MeshWorker::mesh_loop.
+   *
+   * Use numbers::invalid_unsigned_int for @p sub_face_no or @p
+   * sub_face_no_neighbor to indicate that you want to work on the entire face,
+   * not a sub-face.
+   */
+  template <class CellIteratorType>
+  void
+  reinit(const CellIteratorType &cell,
+         const unsigned int &    face_no,
+         const unsigned int &    sub_face_no,
+         const CellIteratorType &cell_neighbor,
+         const unsigned int &    face_no_neighbor,
+         const unsigned int &    sub_face_no_neighbor);
+
+  /**
+   * Re-initialize this object to be used on a interface given by a single face
+   * @p face_no of the cell @p cell. This is useful to use FEInterfaceValues
+   * on boundaries of the domain.
+   *
+   * As a consequence, members like jump() will assume a value of zero for the
+   * values on the "other" side. Note that no sub_face_number is needed as a
+   * boundary face can not neighbor a finer cell.
+   *
+   * After calling this function at_boundary() will return true.
+   */
+  template <class CellIteratorType>
+  void
+  reinit(const CellIteratorType &cell, const unsigned int &face_no);
+
+  /**
+   * Return a reference to the FEFaceValues or FESubfaceValues object
+   * of the specified cell of the interface.
+   *
+   * The @p cell_index is either 0 or 1 and corresponds to the cell index
+   * returned by interface_dof_to_cell_and_dof_index().
+   */
+  const FEFaceValuesBase<dim, spacedim> &
+  get_fe_face_values(const unsigned int cell_index) const;
+
+  /**
+   * Return the vector of JxW values for each quadrature point.
+   *
+   * @note This call requires a prior call to reinit() and having
+   * the @p update_JxW_values flag set in the constructor.
+   */
+  const std::vector<double> &
+  get_JxW_values() const;
+
+  /**
+   * Return the normal vector of the interface in each quadrature point.
+   *
+   * The return value is identical to get_fe_face_values(0).get_normal_vectors()
+   * and therefore, are outside normal vectors from the perspective of the
+   * first cell of this interface.
+   *
+   * @note This call requires a prior call to reinit() and having
+   * the @p update_normal_vectors flag set in the constructor.
+   */
+  const std::vector<Tensor<1, spacedim>> &
+  get_normal_vectors() const;
+
+  /**
+   * Return a reference to the quadrature object in use.
+   */
+  const Quadrature<dim - 1> &
+  get_quadrature() const;
+
+  /**
+   * Return a reference to the quadrature points in real space.
+   *
+   * @note This call requires a prior call to reinit() and having
+   * the @p update_quadrature_points flag set in the constructor.
```

Comment:
We have a macro for this kind of thing: `@dealiiRequiresUpdateFlags{update_quadrature_points}` (see, for example, fe_values.cc, line 2928).